pilot/pkg/bootstrap/server.go

```diff
@@ -1249,3 +1262,27 @@ func (s *Server) addFileWatcher(file string, callback func()) {
 		}
 	}()
 }
+
+func (s *Server) waitForCacheSync(stop <-chan struct{}) bool {
+	if !pilot.EnableWaitCacheSync {
+		return true
```

Comment:
I add this new env as you guys said, though I still think it is just a legacy bug.

Reply:
It's not about features vs bug - but about risks.

Anything risky in the critical path that has not be extensively tested should have a flag.
Or extensive tests.